CMakeLists.txt

```diff
@@ -533,6 +533,24 @@ if(H3_IS_ROOT_PROJECT AND BUILD_TESTING)
         endif()
     endmacro()
 
+    macro(add_h3_cli_test name h3_args expect_string)
+        # Determine the h3 binary name
+        if(WIN32)
+          set(H3CLI ./bin/$<IF:$<CONFIG:Debug>,Debug,Release>/h3.exe)
+        else()
+            set(H3CLI ./bin/h3)
+        endif()
```

Comment:
Can `TARGET_FILE` be used? https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html

Reply:
I will test. I am not super confident, though. (I wish we were still on bare Make, tbh, CMake has too much hidden behavior and irregular grammar)

Reply:
It worked perfectly. Replacing the entirety of the conditional with it, now.